packages/api/lambdas/publish-reports.js

```diff
@@ -227,16 +234,42 @@ async function publishReportSnsMessages(eventMessage, isTerminalStatus, isFailed
  * @returns {Object} - Execution step message or execution input message
  */
 async function getFailedExecutionMessage(inputMessage) {
-  const executionArn = getMessageExecutionArn(inputMessage);
+  try {
+    const executionArn = getMessageExecutionArn(inputMessage);
+    const { events } = await StepFunctions.getExecutionHistory({ executionArn });
+
+    const stepFailedEvents = events.filter(
```

Comment:
I decided that you were right and that the encapsulation wasn't getting us anywhere, so I moved the history parsing directly into this method